mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,34 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(test_fif_fname, preload=True)
+    raw_nopre = Raw(fif_fname, preload=False)
+    raw_eeg_meg = raw.pick_types(meg=True, eeg=True, copy=True)
+    raw_eeg = raw.pick_types(meg=False, eeg=True, copy=True)
+    raw_meg = raw.pick_types(meg=True, eeg=False, copy=True)
+    raw_stim = raw.pick_types(meg=False, eeg=False, stim=True, copy=True)
+    raw_new = raw_meg.append_channels([raw_eeg])
+
+    assert_true([ch in raw_new.ch_names for ch in raw.ch_names])
+    assert_array_equal(raw_new[:, :][0], raw_eeg_meg[:, :][0])
+    assert_array_equal(raw_new[:, :][1], raw[:, :][1])
+    assert_true(all([ch not in raw_new.ch_names for ch in raw_stim.ch_names]))
```

Comment:
I think MRG, no? I mean, I would like it to be MRG :)

On Wed, Jul 15, 2015 at 4:32 PM, Alexandre Gramfort <
notifications@github.com> wrote:

> In mne/io/fiff/tests/test_raw.py
> <https://github.com/mne-tools/mne-python/pull/2302#discussion_r34683905>:
>
> > @@ -115,6 +115,34 @@ def test_copy_append():
> >      assert_equal(data.shape[1], 2 * raw._data.shape[1])
> >
> >
> > +def test_append_channels():
> > +    """Test raw splitting / re-appending channel types
> > +    """
> > +    raw = Raw(test_fif_fname, preload=True)
> > +    raw_nopre = Raw(fif_fname, preload=False)
> > +    raw_eeg_meg = raw.pick_types(meg=True, eeg=True, copy=True)
> > +    raw_eeg = raw.pick_types(meg=False, eeg=True, copy=True)
> > +    raw_meg = raw.pick_types(meg=True, eeg=False, copy=True)
> > +    raw_stim = raw.pick_types(meg=False, eeg=False, stim=True, copy=True)
> > +    raw_new = raw_meg.append_channels([raw_eeg])
> > +
> > +    assert_true([ch in raw_new.ch_names for ch in raw.ch_names])
> > +    assert_array_equal(raw_new[:, :][0], raw_eeg_meg[:, :][0])
> > +    assert_array_equal(raw_new[:, :][1], raw[:, :][1])
> > +    assert_true(all([ch not in raw_new.ch_names for ch in raw_stim.ch_names]))
>
> is this a MRG or WIP PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2302/files#r34683905>.
>